### PR TITLE
Update payment channel state service with latest contract

### PR DESCRIPTION
**Pull request by chetankothari**

### CODE REVIEW

File: packages/core/src/BaseServiceClient.js [hunk 1]
```diff
@@ -59,15 +59,22 @@ class BaseServiceClient {
...
+      { t: 'uint256', v: channelIdStr },
+      { t: 'uint256', v: currentBlockNumber },
+    );
 
     const channelIdBytes = Buffer.alloc(4);
     channelIdBytes.writeUInt32BE(channelId, 0);
 
-    const ChannelStateRequest = this._getChannelStateRequestMethodDescriptor();
-    const channelStateRequest = new ChannelStateRequest();
+    const channelStateRequest = new this.paymentChannelStateServiceClient.getChannelState.requestType();
```
Comment (chetankothari) [inline packages/core/src/BaseServiceClient.js]: Need to revert this as this does not work for web platform.

File: packages/core/src/MPEContract.js [hunk 2]
```diff
@@ -62,108 +64,121 @@ class MPEContract {
...
+    await this._fundEscrowAccount(account, amountInCogs);
 
-    logger.info(`Extending and Funding PaymentChannel[id: ${channelId}] with amount: ${amount} and expiry: ${expiration}`, { tags: ['MPE'] });
+    logger.info(`Extending and Funding PaymentChannel[id: ${channelIdStr}] with amount: ${amount} and expiry: ${expiryStr}`, { tags: ['MPE'] });
     const channelExtendAndAddFundsOperation = this.contract.methods.channelExtendAndAddFunds;
-    return account.sendTransaction(this.address, channelExtendAndAddFundsOperation, channelId, expiration, amount);
+    return account.sendTransaction(this.address, channelExtendAndAddFundsOperation, channelIdStr, expiryStr, amount);
   }
-
+
```
Comment (chetankothari) [inline packages/core/src/MPEContract.js]: Remove extra spacing

File: packages/core/src/MPEContract.js [hunk 3]
```diff
@@ -62,108 +64,121 @@ class MPEContract {
    * @returns {Promise.<TransactionReceipt>}
    */
   async withdraw(account, amountInCogs) {
-    logger.info(`Withdrawing ${amountInCogs}cogs from MPE account`, { tags: ['MPE'] });
+    const amount = new BigNumber(amountInCogs).toFixed();
```
Comment (chetankothari) [inline packages/core/src/MPEContract.js]: Extract into a helper method.

**Comment by vsbogd** [comment]
@chetankothari , could you please explain which changes in contract this PR incorporates.

File: packages/core/src/BaseServiceClient.js [hunk 4]
```diff
@@ -59,7 +59,14 @@ class BaseServiceClient {
...
   async getChannelState(channelId) {
-    const signatureBytes = await this._account.signData({ t: 'uint256', v: channelId });
+    const currentBlockNumber = await this._web3.eth.getBlockNumber();
+    const channelIdStr = new BigNumber(channelId).toFixed();
+    const signatureBytes = await this._account.signData(
+      { t: 'string', v: '__get_channel_state' },
+      { t: 'address', v: this._mpeContract.address },
+      { t: 'uint256', v: channelIdStr },
+      { t: 'uint256', v: currentBlockNumber },
+    );
```
Comment (chetankothari) [inline packages/core/src/BaseServiceClient.js]: @vsbogd the signature format has been changed at the daemon end. This is the change we need to do from the consumer perspective. Rest of the changes are just refactoring.

File: packages/core/src/MPEContract.js (hunk 3 above)
Comment (chetankothari) [reply packages/core/src/MPEContract.js]: Address in [commit](https://github.com/singnet/snet-sdk-js/pull/17/commits/5c2daa487965be1991c316051f66ccc837c9000a)

File: packages/core/src/MPEContract.js (hunk 2 above)
Comment (chetankothari) [reply packages/core/src/MPEContract.js]: Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/17/commits/5c2daa487965be1991c316051f66ccc837c9000a)

File: packages/core/src/BaseServiceClient.js (hunk 1 above)
Comment (chetankothari) [reply packages/core/src/BaseServiceClient.js]: Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/17/commits/41d3d70562fbb29327792853bd628672ecaf0aa9)

File: packages/core/src/BaseServiceClient.js (hunk 4 above)
Comment (vsbogd) [reply packages/core/src/BaseServiceClient.js]: Ah, ok, as far as I see it is contract between daemon and client not Ethereum contract. Thanks.

Comment (chetankothari) [reply packages/core/src/BaseServiceClient.js]: You are right. I think I used contract ambiguously. Sorry for that.

**Pull request merged**